internal/configconverter/loglevel_to_verbosity.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configconverter
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap/zapcore"
+)
+
+type LogLevelToVerbosity struct{}
+
+func (LogLevelToVerbosity) Convert(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return fmt.Errorf("cannot MoveHecTLS on nil *confmap.Conf")
+	}
+
+	const expression = "exporters::logging(/\\w+)?::loglevel"
```

Comment:
I know this is used throughout but there's no word character restriction on component id name values.

Reply:
replaced with `.`